test/test_autocast.py

```diff
@@ -350,11 +352,30 @@ def __exit__(self, exc_type, exc_val, exc_tb):
 
     def test_mps_autocast_error_message(self):
         with self.assertWarnsRegex(
-            UserWarning, "MPS Autocast only supports dtype of torch.float16 currently."
+            UserWarning,
+            "MPS Autocast only supports dtype of torch.bfloat16 and torch.float16 currently.",
         ):
-            with torch.autocast(device_type="mps", dtype=torch.bfloat16):
+            with torch.autocast(device_type="mps", dtype=torch.float32):
                 _ = torch.ones(10)
 
+    # torch.bfloat16 is only supported on macOS 14 and above.
+    @expectedFailureMPSPre14
```

Comment:
Is it okay to use the xfail predicated on version here?

cc @malfet

Reply:
If it works, than sounds good to me :)